src/main/java/nl/ordina/jobcrawler/service/VacancyService.java

```diff
@@ -1,112 +1,108 @@
-        this.skillService = skillService;
     }
 
 
-    //******** Adding ********//
-    // Adding vacancy and accompanying skills and relations to the database //
-    public Vacancy add(Vacancy vacancy) {
-        /**
-         * The function below which is commented out, is used to prevent duplicate skills in the skills table.
-         * For some reason it throws a hibernate exception; detached entity passed to persist
-         * javax.persistence.PersistenceException
-         * We can solve this using an EntityManager. Implementing this will take some time while after the mvp we might put the entire vacancy body into a single column. By commenting out the lines below we allow duplicates.
-         * The unique parameter in Skill.java is set to false. Someone using the application won't notice the difference.
-         */
-        
-        //replacing the skills with existing skills from the database if present
-//        Set<Skill> existingSkills = skillService.linkToExistingSkills(vacancy.getSkills());
-//        existingSkills.forEach(skill -> skill.addVacancy(vacancy)); // add the vacancy to the skill
-//        vacancy.setSkills(existingSkills); // add the skills to the vacancy
-
-
-        if (vacancy.hasValidURL()){    //checking the url, if it is malformed it will throw a VacancyURLMalformedException
-            return vacancyRepository.saveAndFlush(vacancy);
-        } else {
-            throw new VacancyURLMalformedException("Website could not be reached");
-        }
-
-
-    }
-
-    //******** Getting ********//
-    public List<Vacancy> getAllVacancies() {
-        return vacancyRepository.findAll();
-    }
-
-    public Optional<Vacancy> getVacancyByID(UUID id) {
+    /**
+     * Returns the vacancy with the specified id.
+     *
+     * @param id ID of the vacancy to retrieve.
+     * @return An optional of the requested vacancy if found, and an empty optional otherwise.
+     */
+    @Override
+    public Optional<Vacancy> findById(UUID id) {
         return vacancyRepository.findById(id);
     }
 
-    public Set<Vacancy> getVacanciesBySkill(String skill) {
-        return skillService.getVacanciesBySkill(skill);
-    }
 
-    public List<Vacancy> getVacanciesByBroker(String broker) {
-        return vacancyRepository.findByBrokerEquals(broker);
+    /**
+     * Returns all vacancies in the database.
+     *
+     * @return All vacancies in the database.
+     */
+    @Override
+    public List<Vacancy> findAll() {
+        return vacancyRepository.findAll();
     }
 
-    public Optional<Vacancy> getExistingVacancy(String url) {
-        return vacancyRepository.findByVacancyURLEquals(url);
-    }
 
+    /**
+     * Updates the specified vacancy, identified by its id.
+     *
+     * @param vacancy The vacancy to update.
+     * @return True if the update succeeded, otherwise false.
+     */
+    @Override
+    public Vacancy update(UUID id, Vacancy vacancy) {
+        return vacancy;
+    }
```

Comment:
what is method is doing? i think some update code is missing.

Reply:
Correct, we discussed that the vacancies will be immutable as they are automatically scraped from the websites and do not need updating. As this method is implemented from the interface and cannot be removed, I changed it to return ```null``` instead.